Trouble running examples on OS X 10.11.6 (El Capitan)

After being able to build the project on El Capitan (https://github.com/grpc/grpc-web/issues/7), I'm having problems to run the example:

- When I run `make example`, I get a message saying that I should use the `darwin` script. So I interrupted the script, and re-ran the `darwin` script.
- To run `make` from `grpc-web/net/grpc/gateway/examples/echo`, I had to re-install `protoc` (using `brew install protobuf`), but then I got the following error:

```
echo$ make
protoc -I . --cpp_out=. echo.proto
g++ -std=c++11 -I/usr/local/include -I../../../../.. -pthread  -c -o echo.pb.o echo.pb.cc
protoc -I . --grpc_out=. \
  --plugin=protoc-gen-grpc=`which grpc_cpp_plugin` echo.proto
g++ -std=c++11 -I/usr/local/include -I../../../../.. -pthread  -c -o echo.grpc.pb.o echo.grpc.pb.cc
g++ -std=c++11 -I/usr/local/include -I../../../../.. -pthread  -c -o echo_server.o echo_server.cc
g++ echo.pb.o echo.grpc.pb.o echo_server.o -L/usr/local/lib -lgrpc++ -lgrpc -Wl,--no-as-needed -lgrpc++_reflection -Wl,--as-needed -lprotobuf -lpthread -ldl -o echo_server
ld: unknown option: --no-as-needed
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [echo_server] Error 1
```

Am I doing anything wrong?

I happy to try something else, if you have any suggestions.

--- a/net/grpc/gateway/examples/echo/Makefile
+++ b/net/grpc/gateway/examples/echo/Makefile
@@ -33,8 +33,7 @@ ROOT_DIR = ../../../../..
 CXX = g++
 CPPFLAGS += -I/usr/local/include -I$(ROOT_DIR) -pthread
 CXXFLAGS += -std=c++11
-LDFLAGS += -L/usr/local/lib -lgrpc++ -lgrpc                       \
-           -Wl,--no-as-needed -lgrpc++_reflection -Wl,--as-needed \
+LDFLAGS += -L/usr/local/lib -lgrpc++ -lgrpc -lgrpc++_reflection \
            -lprotobuf -lpthread -ldl
 PROTOC = protoc
 PROTOS_PATH = ../..

